### ISSUE TITLE
Support Python 3.7's postponed evaluation of annotations

### ISSUE THREAD

**Issue by PCManticore** [PyCQA/pylint#2069]
### Steps to reproduce
1. Given the following code, run pylint over it:
```
from __future__ import annotations
class C:
    @classmethod
    def from_string(cls, source: str) -> C:
        ...

    def validate_b(self, obj: B) -> bool:
        ...

class B:
    ...
```

### Current behavior
`pylint` is going to emit an `undefined-variable` on Python 3.7 for the annotation of `from_string`.

### Expected behavior
It shouldn't emit that warning as long as we have the `annotations` `__future__` import.

### pylint --version output

master
